src/rws_interface.cpp

```diff
@@ -225,6 +225,97 @@ std::vector<cfg::sys::PresentOption> RWSInterface::getCFGPresentOptions()
   return result;
 }
 
+std::vector<cfg::moc::Single> RWSInterface::getCFGSingles()
+{
+  std::vector<cfg::moc::Single> result;
+
+  RWSClient::RWSResult rws_result = rws_client_.getConfigurationInstances(Identifiers::MOC, Identifiers::SINGLE);
+  if(!rws_result.success) throw std::runtime_error(EXCEPTION_GET_CFG);
+
+  std::vector<Poco::XML::Node*> instances;
+  instances = xmlFindNodes(rws_result.p_xml_document, XMLAttributes::CLASS_CFG_DT_INSTANCE_LI);
+
+  for(size_t i = 0; i < instances.size(); ++i)
+  {
+    std::vector<Poco::XML::Node*> attributes = xmlFindNodes(instances[i], XMLAttributes::CLASS_CFG_IA_T_LI);
+
+    cfg::moc::Single single;
+
+    for(size_t j = 0; j < attributes.size(); ++j)
+    {
+      Poco::XML::Node* attribute = attributes[j];
+      if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, Identifiers::NAME))
+      {
+        single.name = xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE);
+        if(single.name.empty()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "use_single_type"))
+      {
+        single.use_single_type = xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE);
+        if(single.use_single_type.empty()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "use_joint"))
+      {
+        single.use_joint = xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE);
+        if(single.use_joint.empty()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+      }
+      else if(xmlNodeHasAttribute(attribute, Identifiers::TITLE, "base_frame_pos_x"))
+      {
+        std::stringstream ss(xmlFindTextContent(attribute, XMLAttributes::CLASS_VALUE));
+        ss >> single.base_frame.pos.x.value;
+        if(ss.fail()) throw std::runtime_error(EXCEPTION_PARSE_CFG);
+        single.base_frame.pos.x.value *= 1000.0;
```

Comment:
This converts to metres? Would that be something you'd want to do in `abb_librws`? I guess this is done for use with ROS?

If so: I would suggest to not convert here, as it would be at least unexpected for "regular" users (ie: no ROS), and make the ROS code responsible for the conversion.

If you do still want to do this here:

```suggestion
        single.base_frame.pos.x.value *= 1e3;
```

(and similar for the `y` and `z` of course.

Reply:
It converts from meters to millimiters.

The configurations are in meters, but I am reusing [Pose](https://github.com/ros-industrial/abb_librws/blob/582cafeb91b7b90b7cdfd203108d7969a4ed8fbf/include/abb_librws/rws_rapid.h#L593-L616) (i.e. representaion of RAPID pose) to store the data, which has a `pos` field (`x`, `y` and `z` in millimeters).

So it is not related to ROS.

Reply:
Change to scientific notation in 369de2f